services/src/main/java/org/keycloak/protocol/oidc/TokenManager.java

```diff
@@ -199,6 +199,12 @@ public TokenValidation validateToken(KeycloakSession session, UriInfo uriInfo, C
             }
         }
 
+        if (!AuthenticationManager.isClientSessionValid(realm, client, userSession, clientSession)) {
+            logger.debug("Client session is disabled");
+            userSession.removeAuthenticatedClientSessions(Collections.singletonList(client.getId()));
+            throw new OAuthErrorException(OAuthErrorException.INVALID_GRANT, "Client session disabled");
```

Comment:
Nitpick: Isn't it "Client session expired" more accurate than "Client session disabled" ? If yes, similar probably applies also for debug message above.

Reply:
Yep, you are right, I have changed it to `Client session not active` to be consistent with the user session message.